include/arch/riscv/arch/machine.h

```diff
@@ -16,6 +16,24 @@
 #include <arch/sbi.h>
 #include <mode/machine.h>
 
+/* PPTR_BASE must be smaller than KERNEL_ELFBASE. */
+compile_assert(pptr_base_less_elf_base, PPTR_BASE < KERNEL_ELF_BASE_RAW);
+/* physBase must be aligned to a page for verification to succeed. */
+compile_assert(phys_base_page_aligned, IS_ALIGNED(PHYS_BASE_RAW, seL4_PageBits));
+
+/* The following compile time checks are verification artifacts. Not necessarily
+   all systems need to satisfy these, but proofs will need to be changed
+   manually if they are not satisfied. These particular checks also are not
+   necessarily sufficient for all real platforms, but they are good sanity
+   checks to have always on. */
+
+/* PPTR_BASE must be at least twice as far away from KERNEL_ELF_BASE as a LargePage. */
+compile_assert(pptr_base_distance, PPTR_BASE + BIT(seL4_LargePageBits + 1) < KERNEL_ELF_BASE_RAW);
+/* Kernel ELF window must have at least 64k space */
+compile_assert(kernel_elf_distance, KERNEL_ELF_BASE_RAW + BIT(16) < KDEV_BASE);
+/* End of kernel ELF window must not overflow as a uint64 */
+compile_assert(kernel_elf_no_overflow, KERNEL_ELF_BASE_RAW < KERNEL_ELF_BASE_RAW + BIT(16));
```

Comment:
Since this is currently included for both modes (32+64), the default word size is not necessarily 64, i.e. the comment needs to be updated if we leave this here. The check still makes sense on 32 bit systems (checking against 32 bit overflow), so if are happy leaving the check at this location, only the comment needs to change.

Reply:
Have updated the comment to `word_t`.